### PR TITLE
moves the sonata admin description enhancer from resource component

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
fix https://github.com/symfony-cmf/resource/issues/34
move sonata admin description enhancer.

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
Now the php version changes are hitting me.

File: src/Enhancer/ResourceDescriptionEnhancer.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2017 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SonataPhpcrAdminIntegrationBundle\Enhancer;
```
Comment (wouterj) [inline src/Enhancer/ResourceDescriptionEnhancer.php]: I would rather put this in `Description`

File: src/Resources/config/enhancer.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
                 public="false">
             <argument type="service" id="sonata.admin.pool" />
             <argument type="service" id="router" />
-            <tag name="cmf_resource.description.enhancer" alias="sonata_admin" />
+            <tag name="cmf_resource.description.enhancer" alias="sonata_phpcr_admin_integration" />
```
Comment (wouterj) [inline src/Resources/config/enhancer.xml]: I would still call this `sonata_admin`

File: tests/Unit/DependencyInjection/CmfSonataAdminExtensionTest.php [hunk 3]
```diff
@@ -149,12 +149,16 @@ public function testEnhancerExists()
             [
                 'SonataDoctrinePHPCRAdminBundle' => true,
                 'SonataAdminBundle' => true,
+                'JMSSerializerBundle' => true,
```
Comment (wouterj) [inline tests/Unit/DependencyInjection/CmfSonataAdminExtensionTest.php]: this is not needed here

File: src/Enhancer/ResourceDescriptionEnhancer.php (hunk 1 above)
Comment (ElectricMaxxx) [inline src/Enhancer/ResourceDescriptionEnhancer.php]: i was unclear about the naming.

File: tests/Unit/DependencyInjection/CmfSonataAdminExtensionTest.php (hunk 3 above)
Comment (ElectricMaxxx) [inline tests/Unit/DependencyInjection/CmfSonataAdminExtensionTest.php]: but in AppKernel ..

File: src/Resources/config/enhancer.xml (hunk 2 above)
Comment (ElectricMaxxx) [inline src/Resources/config/enhancer.xml]: As the master of resource-bundle still has that alias i got an collision. And when having a orm version with an description enhancer, we will have a collision too.

Comment (ElectricMaxxx) [inline src/Resources/config/enhancer.xml]: so at least i would keep the phpcr somewhere.

File: src/Resources/config/enhancer.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    <services>
+
+        <service
+                id="cmf_sonata_phpcr_admin_integration.description.enhancer"
+                class="Symfony\Cmf\Bundle\SonataPhpcrAdminIntegrationBundle\Description\ResourceEnhancer"
+                public="false">
+            <argument type="service" id="sonata.admin.pool" />
+            <argument type="service" id="router" />
+            <tag name="cmf_resource.description.enhancer" alias="sonata_phpcr_admin_integration" />
```
Comment (wouterj) [inline src/Resources/config/enhancer.xml]: Let's do `sonata_phpcr_admin` then.

File: src/Description/ResourceEnhancer.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Add links and meta-info from Sonata Admin.
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ *
+ * @internal
+ */
+class ResourceEnhancer implements DescriptionEnhancerInterface
```
Comment (wouterj) [inline src/Description/ResourceEnhancer.php]: All description enhancers describe resources. I would call this `SonataEnhancer` or the like, this specific enhancer implementation is about adding things that Sonata knows about our document (links, names, etc.)

**Comment by ElectricMaxxx** [comment]
Changed the requested stuff, squashed it. Ready to merge.

**Pull request merged**